plugins/nsmserver/nsmserver_client.go

```diff
@@ -66,7 +66,29 @@ func (n nsmClientEndpoints) RequestDiscovery(ctx context.Context, cr *nsmconnect
 }
 
 func (n *nsmClientEndpoints) RequestAdvertiseChannel(ctx context.Context, cr *nsmconnect.ChannelAdvertiseRequest) (*nsmconnect.ChannelAdvertiseResponse, error) {
-	return nil, status.Error(codes.InvalidArgument, "Not Implemented...")
+	n.logger.Printf("received Channel advertisement...")
+	for _, c := range cr.NetmeshChannel {
+		n.logger.Infof("For NetworkService: %s channel: %s channel's socket location: %s", c.NetworkServiceName, c.Metadata.Name, c.SocketLocation)
+		nsName := c.NetworkServiceName
+		nsNamespace := "default"
+		if c.Metadata.Namespace != "" {
+			nsNamespace = c.Metadata.Namespace
+		}
+
+		ns := n.objectStore.GetNetworkService(nsName, nsNamespace)
```

Comment:
I like this logic. You're writing this to enable a NSE to add it's channel to an existing NetworkService if it exists, or create it if it doesn't. Seems logical to me.